src/lib.rs

```diff
@@ -115,6 +131,17 @@ pub struct SpanData {
     pub column_end: span::Column<span::OneIndexed>,
 }
 
+#[cfg_attr(feature = "serialize-serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serialize-rustc", derive(RustcDecodable, RustcEncodable))]
+#[derive(Debug, Clone)]
+pub struct CompilationOptions {
+    pub invocation: String,
+    pub crate_name: Option<String>,
+    pub test: bool,
```

Comment:
This seems a bit ad hoc. Would it be better to have a 'target' field or something more generic?